packages/ui/src/views/platform/input-reference-component.tsx

```diff
@@ -0,0 +1,167 @@
+import { ReactNode } from 'react'
+
+import { Button, Caption, ControlGroup, Icon, IconProps, Label } from '@/components'
+import { cn } from '@utils/cn'
+import { cva, type VariantProps } from 'class-variance-authority'
+
+const inputReferenceVariants = cva(
+  'flex min-h-9 cursor-pointer items-center justify-between rounded px-3 py-2 transition-colors',
+  {
+    variants: {
+      variant: {
+        default: 'bg-input-background text-foreground-1',
+        clean: 'bg-transparent'
+      },
+      size: {
+        sm: 'h-8',
+        md: 'h-9'
+      },
+      state: {
+        default: 'border-borders-2 border',
+        disabled: 'bg-background-3 text-foreground-7 border-borders-1 cursor-not-allowed border'
+      }
+    },
+    defaultVariants: {
+      variant: 'default',
+      size: 'md',
+      state: 'default'
+    }
+  }
+)
+
+export interface InputReferenceProps<T> extends VariantProps<typeof inputReferenceVariants> {
```

Comment:
nit: Should `T` be constrained a bit here, to avoid unintended types. This can lead to unexpected behavior if `renderValue` receives a type it doesn't support.

```
export interface InputReferenceProps<T = string> extends ...

Reply:
`T = string` only defaults it to a string, and doesn't really constrain what the user can pass.
I believe typically it'd be a DTO like SecretDTO or ConnectorDTO.